cmd/influxd/launcher/launcher.go

```diff
@@ -410,6 +410,12 @@ func (m *Launcher) run(ctx context.Context) (err error) {
 		return err
 	}
 
+	// use unlimited context for data migration for now.
+	if err := m.kvService.CheckAndMigrate(context.Background()); err != nil {
```

Comment:
What happens if this operation fails dues to programmer error and leaves the user database in a malformed state?